### ISSUE TITLE
Kafka Binder - Consumer Slowdown on certain partitions

### ISSUE THREAD

**Issue by buyukim** [spring-cloud/spring-cloud-stream#2841]
**Describe the issue**
When we upgraded to Spring Boot 3.0 + Spring Cloud Stream 4.0.0 back in July, everything in production ran fine for a day, then during our second overnight where we do a lot of Kafka processing, we noticed significant slowdown, but only on a couple of partitions for a couple topics.  The only way to get things working again was to bounce the instance(s) suffering from the slowdown.

The strange part was that we did not see the expected heartbeat/unhealthy kafka consumer error/warn messages in our logs that I have seen in the past.

Since then, we have since added some Micrometer metrics to measure any I/O being done and have added Dynatrace dashboards so we can see changes over time.  Our Kafka broker has also been upgraded from 2.3 to 2.8 since then.

Were there any default kafka consumer changes between Spring Cloud Stream 3.X and 4.0 (or Spring Kafka)?  Wondering if that could be causing a difference in behavior

**To Reproduce**
Very hard to reproduce, we have a DEV instance of our kafka broker that is on Kafka v2.8, but cannot reproduce via load tests

**Version of the framework**
Spring Cloud Stream 4.0.0

**Expected behavior**

**Screenshots**

**Additional context**

**Comment by sobychacko** [comment]
@buyukim It is very hard to debug without getting more details. I can't think of any particular changes that might cause this. More debugging is needed. I suggest you keep looking at the debug level logs and see if anything suspicious is happening there. If you can provide us a reproducible path, we are more than happy to triage these types of performance issues further.

**Comment by buyukim** [comment]
Yeah, figured as much.  We actually did capture about 15 minutes of debug logging, but haven't seen anything pop out at us
